### PR TITLE
Task 6

### PR DESCRIPTION
**Pull request by ChrisRistoff**

### CODE REVIEW

File: app.js [hunk 1]
```diff
@@ -2,11 +2,12 @@ const express = require("express");
...
 const { sqlErrors, customErrors, serverError } = require("./middleware/errorHandlers");
-const app = express();
-
 const { getDocs } = require("./documentation/docController");
+const { getAllCommentsForArticle } = require("./controllers/commentsController");
+
 const app = express();
 
+
 app.use(express.json());
```
Comment (aloush-dev) [inline app.js]: express.json() is still present even though we are not using it.

File: models/commentsModel.js [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+
+exports.getAllCommentsForArticleModel = async (article_id) => {
+  const comments = await db.query(`
+    SELECT *
+    FROM comments
+    WHERE article_id = $1
+    ORDER BY created_at DESC
+  `, [article_id])
+
+  if (comments.rows.length < 1) return Promise.reject({errCode: 404, errMsg: "Article ID not found"})
```
Comment (aloush-dev) [inline models/commentsModel.js]: This isn't a good method to check if the article ID isn't found.
If there is a valid articleID but no comments we should still send those back.

**Review by aloush-dev — CHANGES_REQUESTED** [review]
I'd like to see a test for what happens if there is a valid ArticleID but no comments associated with it.
Hint it should still be a 200.

Please add this test and address comments and resubmit.
Thanks

**Review by aloush-dev — CHANGES_REQUESTED** [review]
I'd like to see a test for what happens if there is a valid ArticleID but no comments associated with it.
Hint it should still be a 200.

Please add this test and address comments and resubmit.
Thanks

File: models/commentsModel.js [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+
+  if (article.rows.length < 1) return Promise.reject({errCode: 404, errMsg: "Article ID not found"})
+
+  const comments = await db.query(`
+    SELECT *
+    FROM comments
+    WHERE article_id = $1
+    ORDER BY created_at DESC
+  `, [article_id])
+
```
Comment (itsaliabdallah) [inline models/commentsModel.js:16]: Do we need to make 2 database queries here?
Does PSQL not throw an error if the articleID doesn't exist?

File: models/commentsModel.js [hunk 4]
```diff
@@ -0,0 +1,19 @@
+const db = require("../db/connection")
+
+exports.getAllCommentsForArticleModel = async (article_id) => {
+  const article = await db.query(`
+    SELECT article_id FROM articles WHERE article_id = $1
+  `, [article_id])
+
+  if (article.rows.length < 1) return Promise.reject({errCode: 404, errMsg: "Article ID not found"})
```
Comment (aloush-dev) [inline models/commentsModel.js]: Could you reuse a previous model to check if the article exists?

**Review by aloush-dev — APPROVED** [review]
Nice job, Merge when ready.

**Pull request merged**